### PR TITLE
Clean find feature UG

### LINKED ISSUE #296: [BUG] Find Command doesn't follow the general format
* Please refer to activity / patient / doctor on how to format your section

### PR DESCRIPTION
**Pull request by ramaven**
### Summary

* Clean find feature in UG to follow format of rest of UG
resolves #296

### Checks

- [ ] Link PR to issue
- [ ] Add Reviewer
- [ ] Pass CI

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -877,55 +877,77 @@ Format: `clear t/activity`
...
+### [3.4.1 Overview](#table-of-contents)
+The find feature allows you to search for entries based on user input by matching
+the input string with the user-specified fields.
+
+The model that is being searched must be specified too - doctor, patient or activity.
+
+For all find functions, the Keyword is case-insensitive for convenience (“dia” will match diabetic patients even if the user stored the patient's
+  condition as “Diabetes”)
+
+### [3.4.2 Finding a doctor through a keyword `find t/doctor FIELD/[KEYWORDS]`](#table-of-contents)
```
Comment (simonjulianl) [inline docs/UserGuide.md:891]: Btw just find `t/doctor` is fine

File: docs/UserGuide.md [hunk 2]
```diff
@@ -877,55 +877,77 @@ Format: `clear t/activity`
...
+--------------|---------------------------------------------------|
+`n/NAME`      | Matches the full name of the doctor                |
+`p/PHONE_NUMBER`| Matches the phone number of the doctor            | 
+`de/DEPARTMENT` | Matches the department of the doctor           |
+
+**Example:**
+* E.g. `find t/doctor n/Hans Bo` will return doctors whose names are `Hans Gruber`, or `Bo Yang`.
+
+
+### [3.4.3 Finding a patient through a keyword `find t/patient FIELD/[KEYWORDS]`](#table-of-contents)
```
Comment (simonjulianl) [inline docs/UserGuide.md:911]: Similarly for the other headings

File: docs/UserGuide.md [hunk 3]
```diff
@@ -877,55 +877,77 @@ Format: `clear t/activity`
...
-* `n/NAME`: Matches the name field
-* `p/PHONE_NUMBER`: Matches the phone number field
-* `de/DEPARTMENT`: Matches the department field
 
-### [3.4.2 Finding a patient through a keyword `find t/patient FIELD/[KEYWORDS]`](#table-of-contents)
+Parameters    |  Explanation                                      |               
+--------------|---------------------------------------------------|
+`n/NAME`      | Matches the full name of the doctor                |
+`p/PHONE_NUMBER`| Matches the phone number of the doctor            | 
+`de/DEPARTMENT` | Matches the department of the doctor           |
```
Comment (simonjulianl) [inline docs/UserGuide.md:905]: You might want to explain like the constraints are not checked, they only match the value (like you can put any phone number like p/abcde)

**Review by simonjulianl — CHANGES_REQUESTED** [review]
Some comments, also btw, this is still a bug 

image

File: docs/UserGuide.md (hunk 2 above)
Comment (ramaven) [reply docs/UserGuide.md:911]: Ok

File: docs/UserGuide.md (hunk 3 above)
Comment (ramaven) [reply docs/UserGuide.md:910]: Ok done

**Review by simonjulianl — APPROVED** [review]
LGTM

**Pull request merged**